base/util.go

```diff
@@ -1039,7 +1039,25 @@ func InjectJSONProperties(b []byte, kvPairs ...KVPair) (new []byte, err error) {
 
 	kvPairsBytes := make([]KVPairBytes, len(kvPairs))
 	for i, kv := range kvPairs {
-		valBytes, err := JSONMarshal(kv.Val)
+		var valBytes []byte
+		var err error
+
+		switch v := kv.Val.(type) {
+		case int, int8, int16, int32, int64:
+			valBytes = []byte(strconv.Itoa(kv.Val.(int)))
+		case uint, uint8, uint16, uint32, uint64:
+			valBytes = []byte(strconv.FormatUint(v.(uint64), 10))
+		case string:
+			valBytes = []byte(`"` + v + `"`)
+		case bool:
+			valBytes = []byte(strconv.FormatBool(v))
+		case float32:
+			valBytes = []byte(fmt.Sprintf("%v", v))
```

Comment:
Is there a reason to not use %f here?

Reply:
%f has extra trailing 0's which seems like a thing we wouldn't want. I can switch to 'g' which is what 'v' is effectively doing.